### ISSUE TITLE
vmap kwarg handling

### ISSUE THREAD

**Issue by zou3519** [pytorch/functorch#244]
Our kwarg handling seems inconsistent from JAX's:
```
import jax
import jax.numpy as jnp

def f(x, y):
    return x * y

x = jnp.ones([2])
jax.vmap(f)(x, y=x).shape # returns (2,)
```

```
import torch
0import functorch

def f(x, y):
    return x * y

x = torch.ones(2)
functorch.vmap(f)(x, y=x).shape # returns (2, 2)
```.

We should figure out what we want the semantics of kwarg handling to be. Right now it looks like we treat all kwargs as unbatched (and it is documented like that). There are a couple of options:
- we treat all kwargs as batched (this is kind of awkward for values that aren't Tensors)
- we treat all kwargs as unbatched (the current state)
- some mix of both? This is probably too confusing

**Comment by samdow** [comment]
From [JAX docs](https://jax.readthedocs.io/en/latest/jax.html#jax.vmap)
"Arguments passed as keywords are always mapped over their leading axis (i.e. axis index 0)."

To Richard's point, I'm not sure that I personally like this

**Comment by vfdev-5** [comment]
I think mix of both could be reasonable.
Speaking about the current vmap docs:
https://github.com/pytorch/functorch/blob/20c6b32546215885c0179e391950b6c18227b301/functorch/_src/vmap.py#L185-L186

Maybe, it would worth to mention that `args` are Tensors and structures (pytrees) with Tensors. Otherwise, user may think that it can receive any kind of data (no matter vmap use-case) or list of scalars can be batched as well ...
